elastic/kibana
# [Uptime] Refactor es queries and associated components/endpoints

## Summary

The primary goal of this PR was to improve the existing ES queries. There are a lot of tangential changes related to that goal as well. This is also partially the result of breaking changes between HB `6.x` and `7.0`. I also added one component and associated backend code.

## Testing this PR
- Run heartbeat
- Check that:
  - the _overview_ and _monitor_ pages load
  - the you can apply filters on the _overview_ page using the filter buttons
  - the _overview_'s snapshot, monitor list, and error list show up
  - the _monitor page_'s title displays either the monitor name/url, along with the ID
  - the _monitor page_'s charts appear
  - the information in the _monitor_ status bar appears correct
  - you are able to filter the _monitor page_'s check list by `up`/`down`/`all`
